quickstarts/microsoft.compute/vm-cyclecloud/main.bicep

```diff
@@ -0,0 +1,202 @@
+        name: 'HTTPS'
+        properties: {
+          priority: 1020
+          protocol: 'Tcp'
+          access: 'Allow'
+          direction: 'Inbound'
+          sourceAddressPrefix: myIp
+          sourcePortRange: '*'
+          destinationAddressPrefix: '*'
+          destinationPortRange: '443'
+        }
+      }
+    ]
+  }
+}
+
+resource vnet 'Microsoft.Network/virtualNetworks@2020-05-01' = {
+  name: 'cc-vnet'
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        '10.0.0.0/22'
+      ]
+    }
+    subnets: [
+      {
+        name: 'Default'
+        properties: {
+          addressPrefix: '10.0.0.0/24'
+          networkSecurityGroup: {
+            id: nsg.id
+          }
+        }
+      }
+    ]
+  }
+}
+
+resource pip 'Microsoft.Network/publicIpAddresses@2020-05-01' = {
+  name: 'cycleserver-pip'
+  location: location
+  properties: {
+    publicIPAllocationMethod: 'Dynamic'
+  }
+}
+
+resource nic 'Microsoft.Network/networkInterfaces@2020-05-01' = {
+  name: 'cycleserver-nic'
+  location: location
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig'
+        properties: {
+          privateIPAllocationMethod: 'Dynamic'
+          subnet: {
+            id: '${vnet.id}/subnets/Default'
+          }
+          publicIPAddress: {
+            id: pip.id
+          }
+        }
+      }
+    ]
+  }
+}
+
+resource storage 'Microsoft.Storage/storageAccounts@2021-04-01' = {
+  name: saName
+  location: location
+  kind: 'Storage'
+  sku: {
+    name: 'Standard_LRS'
+  }
+}
+
+resource mid 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: 'CycleCloud-MI'
+  location: location
+}
+
+resource vm 'Microsoft.Compute/virtualMachines@2020-06-01' = {
+  name: 'cycleserver'
+  location: location
+  identity: {
+    type: 'UserAssigned'
+    userAssignedIdentities: {
+      '${mid.id}': {}
+    }
+  }
+  properties: {
+    osProfile: {
+      computerName: 'CycleServer'
+      adminUsername: adminUserName
+      customData: base64(customData)
+      linuxConfiguration: {
+        disablePasswordAuthentication: true
+        ssh: {
+          publicKeys: [
+            {
+              path: '/home/${adminUserName}/.ssh/authorized_keys'
+              keyData: csadminSshKey
+            }
+          ]
+        }
+      }
+    }
+    hardwareProfile: {
+      vmSize: vmSize
+    }
+    storageProfile: {
+      imageReference: {
+        publisher: 'OpenLogic'
+        offer: 'CentOS'
+        sku: '8_2'
+        version: 'latest'
+      }
+      osDisk: {
+        name: 'cycleserver-os'
+        createOption: 'FromImage'
+        managedDisk: {
+          storageAccountType: 'StandardSSD_LRS'
+        }
+      }
+    }
+    networkProfile: {
+      networkInterfaces: [
+        {
+          id: nic.id
+        }
+      ]
+    }
+  }
+}
+
+resource rbac 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  name: guid(resourceGroup().id)
```

Comment:
This needs to be:

guid(resourceGroup().id, tenantResourceId(tenantResourceId('Microsoft.Authorization/roleDefinitions', roleDefinitions.contributor), mid.id)